### DISCUSSION TITLE
Proposal for a possible incentive system without launching a new cryptocurrency.

### DISCUSSION THREAD

**Discussion by Destroyinator69420** [freenet/freenet-core#136] [General]
I have noticed that Locutus had no incentive to help the network. I think an incentive system is possible without launching a new cryptocurrency, since as far as I understand Locutus is not a cryptocurrency, but feel free to correct me if I am wrong about that. This incentive system would be based on a simple subtraction problem:

messages you have processed for others - messages others have processed for you
* in this case, you refers to your local node

If the result is less than zero, than all message you request after that you owe to people who processed your messages anyway, and your number will not be positive again until after you have paid back those who loaned their computing power to you. 
Nodes that do not help the network will receive slower service, since in this system, higher trust balances would be processed first, this will reward those who contribute to the network with faster speeds than those who don't.

There are multiple methods to achieve this, one is a trustchain, and another is a distributed hash table, which would sacrifice reliability for lower resource use. However, it will be necessary to use an opaque trustchain similar to Monero if the trustchain is chosen as the store of trust, luckily, ring signatures, stealth addresses, and ringCT are well researched and can be replicated, even if they are not being used to create a cryptocurrency.

If the trustchain is used to store the equation I mentioned, then measures should be implemented to protect the privacy of the trustchain to protect node operators from repercussions of participating in the protocol, since the trustchain will probably last much longer than the messages themselves. It would be smart to flush older blocks once the trustchain is larger than a certain amount, unspent trust tokens in these "wallets" could then be added to the new genesis block. If this design choice is made, I would recommend flushing the trustchain every 90 days.

Links:
https://www.getmonero.org/resources/moneropedia/ringsignatures.html
https://www.getmonero.org/resources/moneropedia/stealthaddress.html
https://www.getmonero.org/resources/moneropedia/ringCT.html

Best Wishes,
Destroyer

**Comment by micwin** [comment]
the more i think of this... the complicater ...

@Destroyinator69420 , your proposal favors local reputation, which I think opens up the discussion (if not already held elsewhere) of wether the net as a whole should favor local high performer over global high performer.

IMHO,if global performers are favored more, this also favors centralization over decentralization, which imho is a bad thing.

A block chain favors global high performer,  so your proposal would be an interesting alternative... BUT

favoring local high performers favors fragmentation of the whole, which might be a problem for consistency, concurrent changes and such.
